<launch>

    <arg name="record_bag" default="true"/>
    <arg name="bag_name" default="20191106_003_for_plots"/>
    <arg name="play_bag" value="true"/>
    <arg name="run_estimator" value="true"/>
    <arg name="run_truth" value="true"/>
    <arg name="display_expected_measurements" default="true"/>

    <group if="$(arg play_bag)">

<!--        Smoother bags=================================================================================-->
<!--        Works decently-->
<!--        <arg name="bag_path" default="/media/humberto/data/camera_imu_calibration_bags/20191008_001.bag"/>-->

<!--        disks motion =================================================================================-->
<!--        <arg name="bag_path" default="/media/humberto/data/camera_imu_calibration_bags/20191015_001.bag"/>-->
<!--        Closer bags====================================================================================-->
<!--        <arg name="bag_path" default="/media/humberto/data/camera_imu_calibration_bags/20191009_001.bag"/>-->
<!--        Tripod bags====================================================================================-->
<!--        <arg name="bag_path" default="/media/humberto/data/camera_imu_calibration_bags/20191017_003.bag"/>-->

<!--        Smooth hand held. This bags started on  20191018    -->
<!--        BAG 20191018_003 is the GLORIUOS BAG ==========================================================-->
<!--        <arg name="bag_path" default="/media/humberto/data/camera_imu_calibration_bags/20191018_003.bag"/>-->
<!--        <arg name="bag_path" default="/media/humberto/data/camera_imu_calibration_bags/20191019_003.bag"/>-->
<!--        <arg name="bag_path" default="/media/humberto/data/camera_imu_calibration_bags/20191021_001.bag"/>-->


<!--        Vector NAV bags started here===================================================================-->
<!--        <arg name="bag_path" default="/media/humberto/data/camera_imu_calibration_bags/vectornav/20191030_001.bag"/>-->
<!--    <arg name="bag_path" default="/media/humberto/data/camera_imu_calibration_bags/vectornav/20191031_002.bag"/>-->
<!--    <arg name="bag_path" default="/media/humberto/data/camera_imu_calibration_bags/vectornav/compensated/20191105_001.bag"/>-->
<!--    <arg name="bag_path" default="/media/humberto/data/camera_imu_calibration_bags/vectornav/compensated/20191106_004.bag"/>-->
    <arg name="bag_path" default="/media/prashant/Samsung_T5/ar_sys_bags/20191106_003.bag"/>

    <node pkg="rosbag" type="play" name="player" output="screen" args="--clock --pause  $(arg bag_path)"/>
    </group>

    <include if="$(arg run_truth)" file="$(find calib_truth)/launch/calib_truth.launch">
    </include>

    <rosparam file="$(find camera_imu_calib)/params/estimator_param.yaml"/>
    <node if="$(arg run_estimator)" name="camera_imu_calibration" pkg="camera_imu_calib" type="camera_imu_calib_node" output="screen">
        <param name="enable_dynamic_update" type="bool" value="true" />
        <param name="use_mocap_to_initialize" type="bool" value="false" />
        <param name="mahalanobis_param" type="double" value="10000000" />
        <param name="number_of_features" type="double" value="16" />
        <param name="publish_full_quaternion" type="bool" value="false" />
        <remap from="/imu/data" to="/imu/imu"></remap>

    </node>

    <group if="$(arg display_expected_measurements)">
        <node name="residual" pkg="camera_imu_calib" type="show_res" output="screen"/>
        <node ns="/" pkg="image_view" type="image_view" name="image_view" output="screen">
            <!--            <remap from="image" to="/camera/rgb/image_raw" />-->
            <remap from="image" to="residual_image" />
            <param name="autosize" type="bool" value="true" />
        </node>
    </group>

<!--    <node if="$(arg record_bag)" name="record" pkg="rosbag" type="record" args="-O $(arg bag_name) ar_single_board/corner ar_single_board/pose ar_single_board/result  imu/data tf_imu/ned/pose_stamped tf_calib/ned/pose_stamped /calib_truth /imu_calib_result"/>-->
    <node if="$(arg record_bag)" name="record" pkg="rosbag" type="record" args="-O $(arg bag_name)  tf_imu/ned/pose_stamped tf_calib/ned/pose_stamped /calib_truth /imu_calib_result expected_measurements ar_single_board/corner"/>

</launch>
